pkg/component/gardener/discoveryserver/deployment.go

```diff
@@ -154,8 +157,39 @@ func (g *gardenerDiscoveryServer) deployment(
 		},
 	}
 
+	injectWorkloadIdentity(deployment, secretNameWorkloadIdentity)
 	utilruntime.Must(gardenerutils.InjectGenericKubeconfig(deployment, secretNameGenericTokenKubeconfig, secretNameVirtualGardenAccess))
 	utilruntime.Must(references.InjectAnnotations(deployment))
 
 	return deployment
 }
+
+func injectWorkloadIdentity(deployment *appsv1.Deployment, secretName string) {
```

Comment:
Not related to this PR, but can you please set `ReadOnly: true` to the other volume mount and `DefaultMode: ptr.To[int32](0400)` to the volume source as well so we keep the consistency?

Suggestion: I would also vote to remove the function and directly set these in the deployment definition since they are not conditional.